src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java

```diff
@@ -567,25 +567,47 @@ private BlockPos getPositionInDirection(final EnumFacing facing, final int dista
     }
 
     /**
-     * Calculates the floor level.
+     * Calculates the floor level
      *
-     * @param position input position.
-     * @return returns BlockPos position with air above.
+     * @param position input position
+     * @return returns BlockPos position with air above
      */
     @NotNull
-    private BlockPos getFloor(@NotNull final BlockPos position)
+    private BlockPos getFloor(@NotNull BlockPos position)
     {
+        final BlockPos floor = getFloor(position, 0);
+        if (floor == null)
+        {
+            return position;
+        }
+        return floor;
+    }
+
+    /**
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) First sentence should end with a period. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AJavadoc_Style)